src/test.js

```diff
@@ -500,6 +509,13 @@ function synchronize( callback, priority ) {
 	}
 }
 
+function randomNumberGenerator( seed ) {
```

Comment:
I'd like to see _this_ be the point where we consume user-specified data, rather than src/core/config.js. And I think it makes sense to reuse [`generateHash`](https://github.com/jquery/qunit/blob/ecf0b36fdde81db9deb122a2feecb75c5080a769/src/test.js#L462) (possibly with modifications) instead of special-casing numbers. For example:
```js
// 32-bit xorshift, requires only a nonzero seed
// http://excamera.com/sphinx/article-xorshift.html
seed = parseInt( generateHash( seed ), 16 ) || -1;
return function() {
	seed ^= seed << 13;
	seed ^= seed >>> 17;
	seed ^= seed << 5;

	// ECMAScript has no unsigned number type
	if ( seed < 0 ) {
		seed += 0x100000000;
	}

	return seed / 0x100000000;
};
```